### DISCUSSION TITLE
How to use this library to write worklist requests and worklist servers

### DISCUSSION THREAD

**Discussion by liaohongxing** [Enet4/dicom-rs#330] [Q&A] [answered]
Is there any example code

**Comment by Enet4** [comment, marked as answer]
The answer would be similar to what [I described in #140](https://github.com/Enet4/dicom-rs/issues/140#issuecomment-878227187). There are [plans](https://github.com/Enet4/dicom-rs/wiki/Roadmap#networking-and-service-class-implementation) for a better abstraction from which new SCUs and SCPs would be constructed. At the moment, implementing new services involves using `dicom_ul` to establish associations and receive/send PDUs. Messages within P-Data fragments can then be encoded using `dicom_object`. [`storescu`](https://github.com/Enet4/dicom-rs/tree/master/storescu) and [`storescp`](https://github.com/Enet4/dicom-rs/tree/master/storescp) provide examples of this.

**Comment by liaohongxing** [comment]
thanks for your quick reply, looking forward to your next work
